tools/src/main/java/org/dashj/platform/tools/WalletTool.java

```diff
@@ -1861,15 +2000,20 @@ private static void dumpDashPay() {
         }
     }
 
-    private static void createUsername(OptionSpec<WaitForEnum> waitForFlag) {
+    private static void createUsername(OptionSpec<WaitForEnum> waitForFlag, String username, Coin credits, boolean useCoinJoin) {
         initializeIdentity();
 
         if (blockchainIdentity == null) {
             try {
-                blockchainIdentity = new BlockchainIdentity(platform, 0, wallet);
 
+                //need to mix
+                if (useCoinJoin &&wallet.getCoinJoinBalance().isLessThan(credits)) {
+                    mix(waitForFlag);
+                }
+
+                blockchainIdentity = new BlockchainIdentity(platform, 0, wallet);
```

Comment:
Before creating the username, we have a mixing step.  Other command line arguments such as `--rounds` and `--sessions` can be used as well to tailer the test.